unittest/commands/test_kim_commands.cpp

```diff
@@ -302,6 +336,133 @@ TEST_F(KimCommandsTest, kim_property)
+                 "keyword=\\[value\\], where value is always an array of one "
+                 "or more comma-separated items.*", 
+                 lmp->input->one("kim_query a0 get_lattice_constant_cubic "
+                                 "crystal"););
+
+    TEST_FAILURE(".*ERROR: Illegal query format.\nInput argument of `"
+                 "crystal=fcc` to kim_query is wrong. The query format is the "
+                 "keyword=\\[value\\], where value is always an array of one "
+                 "or more comma-separated items.*", 
+                 lmp->input->one("kim_query a0 get_lattice_constant_cubic "
+                                 "crystal=fcc"););
+
+    TEST_FAILURE(".*ERROR: Illegal query format.\nInput argument of `"
+                 "crystal=\\[fcc` to kim_query is wrong. The query format is "
+                 "the keyword=\\[value\\], where value is always an array of "
+                 "one or more comma-separated items.*", 
+                 lmp->input->one("kim_query a0 get_lattice_constant_cubic "
+                                 "crystal=[fcc"););
+
+   TEST_FAILURE(".*ERROR: Illegal query format.\nInput argument of `"
+                 "crystal=fcc\\]` to kim_query is wrong. The query format is "
+                 "the keyword=\\[value\\], where value is always an array of "
+                 "one or more comma-separated items.*", 
+                 lmp->input->one("kim_query a0 get_lattice_constant_cubic "
+                                 "crystal=fcc]"););
+    
+    std::string squery("kim_query a0 get_lattice_constant_cubic ");
+    squery += "crystal=[\"fcc\"] species=\"Al\",\"Ni\" units=[\"angstrom\"]";
+
+   TEST_FAILURE(".*ERROR: Illegal query format.\nInput argument of `species="
+                 "\"Al\",\"Ni\"` to kim_query is wrong. The query format is "
+                 "the keyword=\\[value\\], where value is always an array of "
+                 "one or more comma-separated items.*", 
+                 lmp->input->one(squery););
+
+    squery = "kim_query a0 get_lattice_constant_cubic ";
+    squery += "crystal=[\"fcc\"] species=\"Al\",\"Ni\", units=[\"angstrom\"]";
+
+   TEST_FAILURE(".*ERROR: Illegal query format.\nInput argument of `species="
+                 "\"Al\",\"Ni\",` to kim_query is wrong. The query format is "
+                 "the keyword=\\[value\\], where value is always an array of "
+                 "one or more comma-separated items.*", 
+                 lmp->input->one(squery););
+
+    squery = "kim_query a0 get_lattice_constant_cubic crystal=[fcc] "
+             "species=[Al]";
+    TEST_FAILURE(".*ERROR: OpenKIM query failed:.*", lmp->input->one(squery););
+
+    squery = "kim_query a0 get_lattice_constant_cubic crystal=[fcc] "
+             "units=[\"angstrom\"]";
+    TEST_FAILURE(".*ERROR: OpenKIM query failed:.*", lmp->input->one(squery););
+
+    // if (!verbose) ::testing::internal::CaptureStdout();
```

Comment:
what is the deal with the commented out code?

Reply:
@akohlmey the commented ones require a network and special model. Thus they are not general, but I would check them in my update. If you think it is better I can remove them.

Reply:
I would suggest you add an option to `cmake/Modules/Packages/KIM.cmake`, e.g. KIM_EXTRA_UNITTESTS which defaults to OFF and then use that setting to add a corresponding pre-processor define when compiling `test_kim_commands` in `unittest/commands/CMakeLists.txt`. Then you can have the tests included, but only if they are explicitly enabled. The CMake variable should be documented in `doc/src/Build_extras.rst` where it explains what pre-requisites are required.

That way the tests can be easily available to people that have the prerequisites but don't automatically fail with setups that do not have internet access or the required modules installed.

Reply:
@akohlmey sounds great. I would do it. We also have another feature to add which only depends on the new `kim-api` and needs the same pre-processor definition in the `KIM.cmake`. If you agree I will add these two in a new PR  after PR #2488 and the current one are merged into lammps.

Reply:
as soon as this is addressed soon, there is not going to be much of a problem. you just need to be aware that when we see large chunks of commented out code without explanations of it begin added for debugging or otherwise, it will eventually be removed.